processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java

```diff
@@ -609,6 +597,10 @@ public static Dictionary getDictionary(AbsoluteTableIdentifier absoluteTableIden
           blockAssignmentStrategy = BlockAssignmentStrategy.BLOCK_SIZE_FIRST;
         } else {
           blockAssignmentStrategy = BlockAssignmentStrategy.BLOCK_NUM_FIRST;
+          // fall back to BLOCK_NUM_FIRST strategy need to recalculate
+          // the average expected size for each node
+          sizePerNode = calcAvgLoadSizePerNode(blockInfos,uniqueBlocks,
```

Comment:
Avoid calculating times same values. Calculate once and reuse based on the strategy.

Reply:
ok, i modify it.